### DISCUSSION TITLE
fortran iv, fortran-10 v6, ibm7094

### DISCUSSION THREAD

**Discussion by statespacedev** [drforbin/decwar#17] [General]
hi rich - @rcornwell - just a quick note (have been following your work and lots to talk about but will try to keep this concise:) so as part of warming up for working with decwar, have been getting some numerical code from the sixties going on the decwar tops-10 image. thing is, definitely picture running this code on 7094 as well!:) and maybe a cdc setup - but 7094 is no brainer:)

so just leaving a note here - working with the decwar tops-10 image fortran-10 v6 - hopefully will be a smooth hop to a 7094 emulator!:)

here's a previous post related to this

https://github.com/drforbin/decwar/issues/8#issuecomment-2336649682 

current pic is that we're on dec fortran-10 v6 from 1981, which makes sense historically for decwar. this means a dec superset of fortran IV / f66. it meets the f66 ansi standard, and has dec extensions. we know for a fact that v5 did the same and already had the 'include' statement. highly likely that 'include' is one of the dec extensions to f66! here's a nutshell of current learnings for decwar context
dec fortran-10 v5, 1977, fortran iv / f66, plus dec extensions
dec fortran-10 v6, 1981, fortran iv / f66, plus dec extensions
dec fortran-10 v7, 1983, f77, ?plus dec extensions, !probly!?

.r fortra
*^C
.version
FORTRA 6(1144) + 
.r fortra
*/h
FORTRA.HLP -- HELP file for FORTRAN-10 version 6      December 1981

**Comment by statespacedev** [comment]
ahha, and while thinking about it - !note to self!

the decwar utexas teco code for extracting fortran from macro may want / need a particular 'teco version'!?

**Comment by rcornwell** [comment]
Sounds good. When you get it working send me a backup tape image and I will add it to my install scripts.

In terms of the 7094, I doubt that Decwar will ever run on it. First off the machine was single user batch based system. And also had only 32k of memory.

**Reply by statespacedev** [reply]
ahha sorry for the confusion - meant just some little linear algebra fortran programs am working on, here on this tops-10 image - fortran source code originally came from a 7094 - confused things by mixing two topics together (they fit together in my brain, but...:)

am working on a blog post about this fortran code, will have that soon and may help make some sense of all this - 

possible nutshell - dream would be install script(s) could 'make a decwar machine' and 'make a machine with 7094 friendly fortran' - not sure those are even compatible, might be 'and' might be 'or' - just curious!:) what i should probly do is stop talking about 7094 and, if needed, later make multiple versions of this fortran source code - pdp10, 7094, cdc, etc...:) it all should be close to 'fortran66', diffs should be minor...

we have this disk image that is working - for it, merlyn fixed the fortran on the steubentech image - it runs decwar - could i try to convert it to a 'tap file', would that be a backup tape image?

ahha - or maybe better we start with one of your 'baselines', then apply the changes that merlyn made for decwar - basically switching fortran-10 to v6, far as we know so far...?:) i can do testing to check if decwar build and run work:) guess good news is we do have a 'decwar machine' image that works and can help guide us

**Comment by rcornwell** [comment]
I have automatic builds for pretty much all versions of Tops 10. These scripts install the OS, configure the monitor for the hardware, then install all the languages and tools. I have added a few games recently. Would be nice to add Decwar to these installs.

Not sure which versions of Fortran compiler I have on my various installs. ISTR Fortran V6 did not work on 6.03.

My install scripts are actually pretty easy to create. They are basically expect send that executes each command to recovery the files from tap, build, and install the pieces in the correct place.

You would want Decwar source put onto a backup tape. Add in the CMD files to make the build and load easy.

**Comment by statespacedev** [comment]
awesome! yep understood, have been following what you're doing, just never thought of the implications until this very day!:)

let's say this - over the holidays, will target using one of your install scripts to make a decwar machine and install decwar from backup tape - could use this machine for modifying / building / running decwar - what we use merlyn's disk image files for, but in a much more robust sustainable way:)

the good news is that decwar project really orbits around exactly the kind of backup tape of decwar we'll use - the whole reason we're here is that harris newman and merlyn put in the work over the years to get / preserve / use what is effectively the backup/dist tape utexas sent to compuserve in 1982, along with some of the follow on mods compuserve added for making money from paying customers:)

https://en.wikipedia.org/wiki/Decwar 
https://repositories.lib.utexas.edu/items/7e2ccf50-e814-4bce-91d2-a7f6440eabe4 

hi merlyn - @drforbin - this all sound / look / smell ok to you...?

https://en.wikipedia.org/wiki/Decwar#Other_versions 
_Later the CIS (CompuServe Information Service) version of Decwar was released to the public by The University of Texas. This version was highly modified to run on CIS's version of TOPS-10. Merlyn Cousins (Drforbin) later reverse engineered the source back to a form which would run on standard TOPS-10 and simh. This code is archived at the Briscoe Center for American History.[[8]](https://en.wikipedia.org/wiki/Decwar#cite_note-8) The source is also available on GitHub.[[9]](https://en.wikipedia.org/wiki/Decwar#cite_note-9)_
